src/proto/mgmt/storage_query.proto

```diff
@@ -57,6 +57,7 @@ message SmdDevResp {
 	message Device {
 		string uuid = 1; // UUID of blobstore
 		repeated int32 tgt_ids = 2; // VOS target IDs
+		string state = 3; // NORMAL or FAULTY
```

Comment:
Should the state be an enum instead of a string?

Reply:
I wasn't sure about this. There is already an enum in `daos_srv/smd.h`, so it seems like we'd have to maintain two copies of the enum if we did that. In this case I just added the device state to these messages so that we didn't have to make a separate dRPC call down to get it.

Reply:
Would it make sense to make it a uint32 and use CGO to import the possible values from the C? It just seems a little weird to me to use a string to pass around a state with a limited set of possible values.

Reply:
Hmm, yeah. That might work... I'll see what I can do with that.

Reply:
Yeah... So, while I agree that a numeric representation that gets translated for humans when it's displayed is probably the most appropriate way to do this, we have to remember that we don't want to add any linker dependencies on daos/gurt/etc on the dmg side, so we'd have to do this translation in the gRPC handler before sending the string value in the response. I'd like to revisit this in the future if we decide that we want to have the numeric representation in order to do something in response to state changes.